llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtInterfaces.cpp

```diff
@@ -37,12 +37,12 @@ IREE::LinalgExt::detail::verifyLinalgExtOpInterface(Operation *op) {
 
 template <typename Ty, typename DimOpTy>
 static void getDimValues(OpBuilder &b, Location loc, Value v, Ty t,
-                         SmallVector<Value> &dimVals) {
+                         SmallVector<OpFoldResult> &dimVals) {
   for (auto dim : llvm::enumerate(t.getShape())) {
     if (ShapedType::isDynamic(dim.value())) {
-      dimVals.push_back(b.create<DimOpTy>(loc, v, dim.index()));
+      dimVals.push_back((Value)b.create<DimOpTy>(loc, v, dim.index()));
```

Comment:
Should we use something like `.getResult()` here?

Reply:
This kind of casting does not follow MLIR and IREE convention. I never see code like this. Does one of below snippet work?

```suggestion
      dimVals.push_back(b.create<DimOpTy>(loc, v, dim.index()).getResult());
```

```suggestion
      dimVals.push_back(b.createOrFold<DimOpTy>(loc, v, dim.index()));
```

I thought your change is equivalent to the former one.

Reply:
Sorry yes, .getResult() works. Not sure what happened earlier when I was seeing a compile error.